mlx/primitives.h

```diff
@@ -677,6 +677,47 @@ class Convolution : public UnaryPrimitive {
   void eval(const std::vector<array>& inputs, array& out);
 };
 
+class ConvolutionTranspose : public UnaryPrimitive {
+ public:
+  explicit ConvolutionTranspose(
```

Comment:
What's the rationale for making a new primitve for this? Why not just use the Convolution primitive as is?

Reply:
I did that to implement the different `vjp` method, but realized that leads to some code duplication. I can extend the `Convolution` primitive to avoid that.

Reply:
Done.